File: MessageList/app/src/main/java/ru/startandroid/messagelist/app/dagger/AppComponent.java
package ru.startandroid.messagelist.app.dagger;

import dagger.Component;
import ru.startandroid.messagelist.app.dagger.module.AppModule;
import ru.startandroid.messagelist.app.dagger.module.AppSubComponentsModule;
import ru.startandroid.messagelist.app.dagger.module.DataModule;
import ru.startandroid.messagelist.app.dagger.scope.AppScope;
import ru.startandroid.messagelist.domain.messages.MessageRepository;

@AppScope
@Component(modules = {AppModule.class, AppSubComponentsModule.class, DataModule.class})
public interface AppComponent {
    void injectComponentsHolder(ComponentHolder componentsHolder);
    MessageRepository getMessageRepository();
}

File: Vocabulary/app/src/main/java/ru/startandroid/vocabulary/app/dagger/AppComponent.kt
package ru.startandroid.vocabulary.app.dagger

import dagger.Component
import ru.startandroid.vocabulary.app.ComponentHolder

@AppScope
@Component(modules = arrayOf(AppModule::class, AppSubComponentsModule::class, DataModule::class, UseCaseModule::class))
interface AppComponent {
    fun injectComponentsHolder(componentHolder: ComponentHolder) : Unit
}